back/engines/commercial/report_builder/spec/services/report_builder/queries/analytics/active_users_spec.rb

```diff
@@ -6,23 +6,39 @@
   subject(:query) { described_class.new(build(:user)) }
 
   describe '#run_query' do
+    before do
+      FactoryBot.rewind_sequences
+      @date_september = Date.new(2022, 9, 10)
+      @date_october = Date.new(2022, 10, 5)
+
+      @dimension_date_sept = create(:dimension_date, date: @date_september)
+      @dimension_date_oct = create(:dimension_date, date: @date_october)
+      create(:dimension_type, name: 'idea', parent: 'post')
+      create(:dimension_type, name: 'comment', parent: 'idea')
+    end
+
     it 'returns active users' do
-      date = Date.new(2022, 9, 1)
       project = create(:single_phase_ideation_project)
-      create(:dimension_date, date: date)
-      create(:dimension_type, name: 'idea', parent: 'post')
-      create(:idea, created_at: date, project: project)
+      create(:idea, created_at: @date_september, project: project)
 
-      params = { start_at: date - 1.day, end_at: date + 1.day, project_id: project.id }
+      params = {
+        start_at: (@date_september - 1.day).to_s,
+        end_at: (@date_september + 1.day).to_s,
+        project_id: project.id
+      }
       expect(query.run_query(**params)).to eq(
         [
           [{
             'count_participant_id' => 1,
             'dimension_date_created.month' => '2022-09',
-            'first_dimension_date_created_date' => date
+            'first_dimension_date_created_date' => @date_september
           }],
           [{
             'count_participant_id' => 1
+          }],
+          [],
```

Comment:
Yeah, it's a quirk of the analytics implementation. If there are no records, an empty array if given. But (IIRC) if there are records, but let's say the filters make it so that you still end up with no matches, then you get a `0`